generic3g/tests/gridcomps/FakeDynGridComp.F90

```diff
@@ -0,0 +1,155 @@
+#include "MAPL_Generic.h"
+
+module mapl3g_FakeDynGridComp
+
+   use mapl_ErrorHandling
+   use generic3g
+   use mapl3g_FieldCondensedArray, only: assign_fptr_condensed_array
+   use mapl3g_Field_API, only: MAPL_FieldEmptyComplete, VerticalStaggerLoc, VERTICAL_STAGGER_CENTER
+   use mapl3g_UngriddedDims
+   use esmf
+
+   implicit none
+   private
+
+   public :: SetServices
+
+contains
+
+   subroutine SetServices(gridcomp, rc)
+      type(ESMF_GridComp) :: gridcomp
+      integer, intent(out) :: rc
+
+      integer :: status
+
+      ! Set entry points
+      call MAPL_GridCompSetEntryPoint(gridcomp, ESMF_METHOD_INITIALIZE, init, phase_name="GENERIC::INIT_REALIZE", _RC)
+      call MAPL_GridCompSetEntryPoint(gridcomp, ESMF_METHOD_RUN, run, phase_name="run", _RC)
+
+      _RETURN(_SUCCESS)
+   end subroutine setServices
+
+   subroutine init(gridcomp, importState, exportState, clock, rc)
+      type(ESMF_GridComp) :: gridcomp
+      type(ESMF_State) :: importState
+      type(ESMF_State) :: exportState
+      type(ESMF_Clock) :: clock
+      integer, intent(out)  :: rc
+
+      type(ESMF_Geom) :: geom
+      type(ESMF_Field) :: field1, field2
+      class(VerticalGrid), allocatable :: vertical_grid
+      integer :: num_levels, status
+
+      call MAPL_GridCompGet(gridcomp, geom=geom, vertical_grid=vertical_grid, _RC)
+      num_levels = vertical_grid%get_num_levels()
+
+      call ESMF_StateGet(exportState, "PL", field1, _RC)
+      call field_complete_(field1, geom, num_levels, VERTICAL_STAGGER_CENTER, "hPa", "air_pressure", rc)
+      call set_pressure_(field1, _RC)
+
+      call ESMF_StateGet(exportState, "T_DYN", field2, _RC)
+      call field_complete_(field2, geom, num_levels, VERTICAL_STAGGER_CENTER, "K", "temeperature", rc)
+      call set_temperature_(field2, _RC)
+
+      _RETURN(_SUCCESS)
+      _UNUSED_DUMMY(importState)
+      _UNUSED_DUMMY(clock)
+   end subroutine init
+
+   subroutine run(gridcomp, importState, exportState, clock, rc)
+      type(ESMF_GridComp) :: gridcomp
+      type(ESMF_State) :: importState
+      type(ESMF_State) :: exportState
+      type(ESMF_Clock) :: clock
+      integer, intent(out) :: rc
+
+      integer :: status
+
+      call MAPL_RunChildren(gridcomp, phase_name="run", _RC)
+
+      _RETURN(_SUCCESS)
+      _UNUSED_DUMMY(importState)
+      _UNUSED_DUMMY(exportState)
+      _UNUSED_DUMMY(clock)
+   end subroutine run
+
+   subroutine field_complete_(field, geom, num_levels, vertical_stagger, units, standard_name, rc)
```

Comment:
I do not understand why this is needed.      User gridcomps should let the generic layer allocate fields.    the fact that this is needed here suggests a hole in the vgrid functionality that should be addressed.

Reply:
Made the changes
- Linear interpolation matrix between src and dst vcoord are now done in the `update` step.
- That change lets us have the FakeDyn gridcomp's initialize step run during the `INIT_USER` phase where the fields are complete - so no need to allocate them.

Now we don't need to retrieve `vertical_grid` from the gridcomp (via `MAPL_GridCompGet`, but it may not be a bad idea to leave that capability in.